src/mozanalysis/experiment.py

```diff
@@ -0,0 +1,545 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from pyspark.sql import functions as F
+
+from mozanalysis.utils import add_days
+
+
+class Experiment(object):
+    """Get DataFrames of experiment data; store experiment metadata.
+
+    The methods here query data in a way compatible with the following
+    principles, which are important for experiment analysis:
+
+    - The population of clients in each branch must have the same
+        properties, aside from the intervention itself and its
+        consequences; i.e. there must be no underlying bias in the
+        branch populations.
+    - Every client should have an equal opportunity to convert, so that
```

Comment:
what do we mean by convert here?

Reply:
I have a bias towards binary metrics, where we measure whether or not a user "converts" (does the thing of interest) during a particular period of time (the conversion window, e.g. "the first week after they enroll"). So that's what I meant, but I agree it doesn't make sense in this more general context. I'll try to think of something that fits better - have you got any ideas?